### ISSUE TITLE
"history_audit" does not exist on database upgrade 

### ISSUE THREAD

**Issue by shenweiyan** [galaxyproject/galaxy#19016]
**Describe the bug**
I upgraded the postgresql database from 20.09 to 24.0 in the local galaxy instance. Executing `sh manage_db.sh upgrade` showed that everything was normal, but when I executed `sh run.sh` to start Galaxy, I received the following prompt. How can I solve this problem? Thank you.

I use psql (PostgreSQL) 10.3.

```bash
galaxy.webapps.galaxy.controllers.user DEBUG 2024-10-17 09:17:13,544 [pN:main.1,p:27733,tN:WSGI_1] trans.app.config.auth_config_file: /galaxy-dist/galaxy/config/auth_conf.xml
galaxy.auth.providers.localdb DEBUG 2024-10-17 09:17:13,631 [pN:main.1,p:27733,tN:WSGI_1] User: galaxy@ribobio.com, LOCALDB: True
galaxy.web.framework.decorators ERROR 2024-10-17 09:17:13,662 [pN:main.1,p:27733,tN:WSGI_1] Uncaught exception in exposed API method:
Traceback (most recent call last):
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1910, in _execute_context
    self.dialect.do_execute(
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.UndefinedTable: relation "history_audit" does not exist
LINE 2: FROM history_audit
             ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/galaxy-dist/galaxy/lib/galaxy/web/framework/decorators.py", line 346, in decorator
    rval = func(self, trans, *args, **kwargs)
  File "/galaxy-dist/galaxy/lib/galaxy/webapps/galaxy/controllers/user.py", line 143, in login
    return self.__validate_login(trans, payload, **kwd)
  File "/galaxy-dist/galaxy/lib/galaxy/webapps/galaxy/controllers/user.py", line 200, in __validate_login
    trans.handle_user_login(user)
  File "/galaxy-dist/galaxy/lib/galaxy/webapps/base/webapp.py", line 849, in handle_user_login
    self._associate_user_history(user, prev_galaxy_session)
  File "/galaxy-dist/galaxy/lib/galaxy/webapps/base/webapp.py", line 813, in _associate_user_history
    and users_last_session.current_history
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/orm/attributes.py", line 487, in __get__
    return self.impl.get(state, dict_)
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/orm/attributes.py", line 959, in get
    value = self._fire_loader_callables(state, key, passive)
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/orm/attributes.py", line 995, in _fire_loader_callables
    return self.callable_(state, passive)
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/orm/strategies.py", line 912, in _load_for_state
    return self._emit_lazyload(
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/orm/strategies.py", line 996, in _emit_lazyload
    return loading.load_on_pk_identity(
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/orm/loading.py", line 530, in load_on_pk_identity
    session.execute(
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 1717, in execute
    result = conn._execute_20(statement, params or {}, execution_options)
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1710, in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/sql/elements.py", line 334, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1577, in _execute_clauseelement
    ret = self._execute_context(
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1953, in _execute_context
    self._handle_dbapi_exception(
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 2134, in _handle_dbapi_exception
    util.raise_(
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1910, in _execute_context
    self.dialect.do_execute(
  File "/galaxy-dist/galaxy/.venv/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedTable) relation "history_audit" does not exist
LINE 2: FROM history_audit
             ^

[SQL: SELECT history.update_time AS history_update_time, (SELECT max(history_audit.update_time) AS max_1
FROM history_audit
WHERE history_audit.history_id = history.id) AS anon_1, history.id AS history_id, history.create_time AS history_create_time, history.user_id AS history_user_id, history.name AS history_name, history.hid_counter AS history_hid_counter, history.deleted AS history_deleted, history.purged AS history_purged, history.importing AS history_importing, history.genome_build AS history_genome_build, history.importable AS history_importable, history.slug AS history_slug, history.published AS history_published, history.preferred_object_store_id AS history_preferred_object_store_id, history.archived AS history_archived, history.archive_export_id AS history_archive_export_id
FROM history
WHERE history.id = %(pk_1)s]
[parameters: {'pk_1': 1119}]
(Background on this error at: https://sqlalche.me/e/14/f405)
```

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: 24.0
Commit: 03457dff5caac5869a7028812eb6a35d8e506b92

**Browser and Operating System**
Operating System: CentOS Linux release 7.7.1908 (Core)
Browser: Firefox, Chrome, Chrome-based, Safari

**To Reproduce**
`sh manage_db.sh upgrade`

**Expected behavior**

**Screenshots**

**Additional context**
psql (PostgreSQL) 10.3

**Comment by bgruening** [comment]
@shenweiyan have you update in one go or have you gone through all DB releases step by step?

**Comment by shenweiyan** [comment]
> sh manage_db.sh upgrade

I performed the update in one go, but the correct approach should be to go through all DB releases step by step? For example:
```bash
sh manage_db.sh upgrade 21.01  
sh manage_db.sh upgrade 21.05  
....  
sh manage_db.sh upgrade 24.1
```

**Comment by shenweiyan** [comment]
Thank you very much for everyone's help. My problem has been solved. Here is the solution.

1. I noticed that the "history_audit" database table was added in the release_21.05 version (#11914). So I cloned the release_21.05 repo and then built the `.venv` environment.

2. After the `.venv` environment and requirements are installed, execute `sh manage_db.sh upgrade` to successfully upgrade the postgresql database from 20.09 to 21.05. At this point, the newly added "history_audit" database table can be seen in the upgraded data backend.

3. Finally, upgrade the database from version 21.05 to version 24.0.
